### ISSUE TITLE
Add ability to symlink to source file rather than copy/move

### ISSUE THREAD

**Issue by ssteinerx** [marvinlabs/customer-area#99]
I would like to be able to add symlinks to files rather than have to copy/move them from the "ftp" folder.

**Comment by vpratfr** [comment]
@ssteinerx by default we should keep the 'move' method. However, I agree that we could implement another method to create symlinks. Feel free to submit a PR for this.

Could look like (pseudo-code):

```
$move_method = apply_filters('cuar/core/files/move-method', 'move');
switch ($move_method) {
  case 'move: // ...
    break;
  case 'copy': // ...
    break;
  case 'symlink': // ...
    break;
  default:
    wp_die('This move method is invalid');
    // This is important, just fail blindly
}
```

[This should be the function to change](https://github.com/marvinlabs/customer-area/blob/f8efe3b2c53195c361e0e34b0c3f855374c855c8/src/php/core-addons/private-file/private-file-addon.class.php#L362)

Then in the admin interface, instead of a checkbox to say "delete file after copy", we could have a dropdown to choose one of the methods.

**Comment by vpratfr** [comment]
@ssteinerx By the way, I am working on a client project too currently so I won't have much time to merge your PRs/correct bugs. Should get more time at the end of next week.

**Comment by ssteinerx** [comment]
I have added the symlink functionality to the sync plugin.  Easy to merge back into the main plugin later.

I hesitate to pile more changes into my feature/insert-functions branch until it's merged back into the development branch as it's really getting to be quite a lot for one PR.  I'd like to see the main customer-area plugin stabilized and released with the insert-functions merged in before I launch my client's site as I'm running it on my  branch right now and that makes me a little nervous...

I'll just leave it in the sync plugin for now and make a feature/symlink branch after everything's merged whenever you have time.

**Comment by vpratfr** [comment]
Kind of "symlink" is possible with the new bullk creation function available since 6.3. See function phpdoc comments.
